src/Components/shareList.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { fb } from '../lib/firebase';
+import { setLocalToken } from '../lib/token.js';
+
+class ShareList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { token: '' };
+
+    this.updateInput = this.updateInput.bind(this);
```

Comment:
So glad you brought up this pattern 👀.

You can get the same result by writing:

```js
class ShareList extends React.Component {

  constructor(props) {
    super(props)
    this.state = { token: ''};
  }

  updateInput = (event) => {
    this.setState({....})
  }

  // More code omitted for brevity
}
```

And you can go even further:

```js
class ShareList extends React.Component {

  state = { token: '' }

  updateInput = (event) => {
    this.setState({....})
  }

  // More code omitted for brevity
}
```

![](https://media.giphy.com/media/2rqEdFfkMzXmo/giphy.gif)

Reply:
Part of the reason why I like to stray away from class based components is the fact that we have to use `this`, and also manage `this` like such. I'd prefer to just to use JavaScript closures as intended with functional components, and not worry about `this` ever!

```
function ShareList() {
  const [token, setToken] = useState();

  function updateInput(e) {
    setToken(e.currentTarget.value)
  }

  return <input onChange={updateInput}>
}
```